query/src/main/scala/filodb/query/exec/PeriodicSamplesMapper.scala

```diff
@@ -400,11 +394,7 @@ class SlidingWindowIterator(raw: RangeVectorCursor,
       false
     } else {
       val headIsOutsideWindow = windowQueue.head.timestamp <= curWindowStart
-      // One of the following two conditions need to hold true:
-      // 1. if no need for last sample, and head is outside the window
-      (!rangeFunction.needsLastSample && headIsOutsideWindow) ||
-        // 2. if needs last sample, then ok to remove window's head only if there is more than one item in window
-        (rangeFunction.needsLastSample && windowQueue.size > 1 && headIsOutsideWindow)
+      headIsOutsideWindow
```

Comment:
One liner:
`windowQueue.nonEmpty && windowQueue.head.timestamp <= curWindowStart`

Can make it inline condition

Reply:
ack